File: Tutorial5-3aDagger-Android/src/main/java/com/example/tutorial5_1dagger_android/MyApplication.java
package com.example.tutorial5_1dagger_android;


import com.example.tutorial5_1dagger_android.di.DaggerApplicationComponent;

import dagger.android.AndroidInjector;

public class MyApplication extends dagger.android.support.DaggerApplication {


    @Override
    public void onCreate() {
        super.onCreate();
    }

    @Override
    protected AndroidInjector<? extends dagger.android.support.DaggerApplication> applicationInjector() {
        return DaggerApplicationComponent.builder().application(this).build();
    }
}

File: Tutorial5-2Dagger-Android-SubComponents2SeedInstance/src/main/java/com/example/tutorial5_2dagger_android_subcomponents2/MyApplication.kt
package com.example.tutorial5_2dagger_android_subcomponents2

import com.example.tutorial5_2dagger_android_subcomponents2.di.DaggerApplicationComponent
import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication

class MyApplication : DaggerApplication() {


    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        return DaggerApplicationComponent.builder().application(this).build()
    }
}